javalib/src/main/scala/java/util/stream/StreamImpl.scala

```diff
@@ -496,14 +516,51 @@ private[stream] class StreamImpl[T](
       coercedPriorStages
     )
       .asInstanceOf[DoubleStream]
+  }
+
+  def mapToInt(mapper: ToIntFunction[_ >: T]): IntStream = {
+    commenceOperation()
 
+    val spl = new Spliterators.AbstractIntSpliterator(
+      _spliter.estimateSize(),
+      _spliter.characteristics()
+    ) {
+      def tryAdvance(action: IntConsumer): Boolean =
+        _spliter.tryAdvance((e: T) => action.accept(mapper.applyAsInt(e)))
+    }
+
+    val coercedPriorStages = pipeline
+      .asInstanceOf[ArrayDeque[IntStreamImpl]]
```

Comment:
Is it possible that `pipeline` would contain `IntStream` element which is not an instance of `IntStreamImpl`? For example, if somebody implements a custom `IntStream`, can a custom implementation be added to pipeline? It seems to be possible during `flatMap`-like operations. In such case it would fail at runtime.